db/models/articles.model.js

```diff
@@ -36,9 +36,18 @@ exports.fetchAllArticles = (sortBy = "created_at", order = "DESC") => {
 		return Promise.reject({ status: 400, message: "Invalid order query" });
 	}
 	let sqlString =
-		"SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON articles.article_id=comments.article_id GROUP BY articles.article_id";
-	sqlString += ` ORDER BY ${sortBy} ${order};`;
+		"SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON articles.article_id=comments.article_id";
+	if (topic) {
+		sqlString += ` WHERE articles.topic = '${topic}'`;
+	}
+	sqlString += ` GROUP BY articles.article_id ORDER BY ${sortBy} ${order};`;
 	return db.query(sqlString).then(({ rows }) => {
+		if (rows.length === 0) {
+			return Promise.reject({
+				status: 404,
+				message: "No articles found on that topic",
+			});
```

Comment:
You may need to do a separate checkTopicExists check because an empty rows array is not necessarily an error 👍